cvat-ui/src/actions/settings-actions.ts

```diff
@@ -409,3 +415,110 @@ export function resetImageFilters(): AnyAction {
+                    }
+                    default: {
+                        break;
+                    }
+                }
+            }
+        }
+
+        dispatch(setSettings(newSettings));
+        if ('shortcuts' in loadedSettings) {
+            const updateKeyMap = structuredClone(shortcuts.keyMap);
+            for (const key of Object.keys(loadedSettings.shortcuts.keyMap)) {
+                const value = loadedSettings.shortcuts.keyMap[key];
+                if (key in updateKeyMap) {
+                    updateKeyMap[key].sequences = value.sequences;
+                }
+            }
+            for (const key of Object.keys(updateKeyMap)) {
+                const currValue = {
+                    [key]: { ...updateKeyMap[key] },
+                };
+                const conflictingShortcuts = conflictDetector(currValue, shortcuts.keyMap);
+                if (conflictingShortcuts) {
+                    for (const conflictingShortcut of Object.keys(conflictingShortcuts)) {
+                        for (const sequence of currValue[key].sequences) {
+                            for (const conflictingSequence of conflictingShortcuts[conflictingShortcut].sequences) {
+                                if (conflict(sequence, conflictingSequence)) {
+                                    updateKeyMap[conflictingShortcut].sequences = [
+                                        ...updateKeyMap[conflictingShortcut].sequences.filter(
+                                            (s: string) => s !== conflictingSequence,
+                                        ),
+                                    ];
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            dispatch(shortcutsActions.registerShortcuts(updateKeyMap));
+        }
+    };
+}
+
+export function updateCachedSettings(settings: CombinedState['settings'], shortcuts: CombinedState['shortcuts']) {
+    const settingsForSaving: any = {
+        shortcuts: {
+            keyMap: {},
+        },
+        imageFilters: [],
+    };
+    const supportedImageFilters = [ImageFilterAlias.GAMMA_CORRECTION];
+    for (const [key, value] of Object.entries(settings)) {
+        if (['player', 'workspace'].includes(key)) {
+            settingsForSaving[key] = value;
+        }
+        if (key === 'imageFilters') {
+            const filters = [];
+            for (const filter of value) {
+                if (supportedImageFilters.includes(filter.alias)) {
+                    filters.push(filter.modifier.toJSON());
+                }
+            }
+            settingsForSaving.imageFilters = filters;
+        }
+    }
+    for (const [key] of Object.entries(shortcuts.keyMap)) {
+        if (key in shortcuts.defaultState) {
+            settingsForSaving.shortcuts.keyMap[key] = {
+                sequences: shortcuts.keyMap[key].sequences,
+            };
+        }
+    }
```

Comment:
What do you think about some refactoring?

```suggestion
    const supportedImageFilters = [ImageFilterAlias.GAMMA_CORRECTION];
    const settingsForSaving = {
        player: settings.player,
        workspace: settings.workspace,
        shortcuts: {
            keyMap: Object.entries(shortcuts.keyMap).reduce(() => { ...TODO ...}),
        },
        imageFilters: settings.imageFilters.filter((imageFilter) => supportedImageFilters.includes(filter.alias)).map((imageFilter) => imageFilter.modifier.toJSON()),
    };
```

Reply:
Yea, looks better